tests/unit/withOnyxTest.js

```diff
@@ -76,6 +77,34 @@ describe('withOnyx', () => {
             });
     });
 
+    it('should batch correctly together little khachapuris', () => Onyx.mergeCollection(ONYX_KEYS.COLLECTION.TEST_KEY, {
+        [`${ONYX_KEYS.COLLECTION.TEST_KEY}1`]: {ID: 999},
+    }).then(() => Onyx.merge(ONYX_KEYS.SIMPLE_KEY, 'prev_string')).then(() => Onyx.merge(ONYX_KEYS.SIMPLE_KEY_2, 'prev_string2')).then(() => {
+        const TestComponentWithOnyx = withOnyx({
+            testKey: {
+                key: `${ONYX_KEYS.COLLECTION.TEST_KEY}1`,
+            },
+            simpleKey: {
+                key: ONYX_KEYS.SIMPLE_KEY,
+            },
+            simpleKey2: {
+                key: ONYX_KEYS.SIMPLE_KEY_2,
+            },
+        })(ViewWithObject);
+        const onRender = jest.fn();
+        render(<TestComponentWithOnyx onRender={onRender} />);
+
+        return waitForPromisesToResolve()
+            .then(() => {
+                Onyx.mergeCollection(ONYX_KEYS.COLLECTION.TEST_KEY, {test_1: {ID: 123}});
+                Onyx.merge(ONYX_KEYS.SIMPLE_KEY, 'string');
+                return Onyx.merge(ONYX_KEYS.SIMPLE_KEY_2, 'string2');
+            })
+            .then(() => {
+                expect(onRender).toHaveBeenCalledTimes(2);
```

Comment:
Could you please add a comment to explain why it's expected to be called exactly two times?

Reply:
Sure, just a sec

Reply:
done